src/controllers/project.controller.ts

```diff
@@ -300,18 +309,91 @@ export class ProjectController {
 			this.project = project;
 			this.view.show(this.project.name);
 
-			this.notifyViewResourcesPathChanged();
-
 			this.startInterval(
 				ProjectIntervalTypes.deployments,
 				() => this.loadAndDisplayDeployments(),
 				this.deploymentsRefreshRate
 			);
+
+			const isResourcesPathExist = await this.getResourcesPathFromContext();
+
+			if (isResourcesPathExist) {
+				this.notifyViewResourcesPathChanged();
+				return;
+			}
+
+			const { data: existingResourcesToDownload } = await this.getResourcesPathToDownload();
+
+			if (!existingResourcesToDownload) {
+				this.notifyViewResourcesPathChanged();
+				return;
+			}
+
+			const userResponse = await this.promptUserToDownloadResources();
+			if (userResponse !== translate().t("projects.downloadResourcesDirectoryApprove")) {
+				return;
+			}
+			const newLocalResourcesPath = await this.promptUserForResourcesLocalDirectory();
+
+			if (!newLocalResourcesPath || !newLocalResourcesPath.length) {
+				commands.executeCommand(
+					vsCommands.showWarnMessage,
+					translate().t("projects.downloadResourcesDirectoryNotSelected")
+				);
+				return;
+			}
+			this.downloadProjectResources(newLocalResourcesPath, existingResourcesToDownload!);
+			await commands.executeCommand(vsCommands.setContext, this.projectId, { path: newLocalResourcesPath[0].path });
```

Comment:
If we choose directory, then every array item should have the same `path`, so taking it from the first item, shouldn't bother us.